MbDotNet/MountebankClient.cs

```diff
@@ -65,35 +61,45 @@ public TcpImposter CreateTcpImposter(int port, string name = null, TcpMode mode
         public void DeleteImposter(int port)
         {
             var imposter = Imposters.FirstOrDefault(imp => imp.Port == port);
+            _requestProxy.DeleteImposter(port);
 
             if (imposter != null)
             {
-                _requestProxy.DeleteImposter(port);
                 Imposters.Remove(imposter);
             }
         }
 
         /// <summary>
-        /// Deletes all imposters from mountebank. Will also remove the imposters from the collection
+        /// Deletes all imposters from mountebank. Will also remove the imposter from the collection
         /// of imposters that the client maintains.
         /// </summary>
         public void DeleteAllImposters()
         {
             _requestProxy.DeleteAllImposters();
             Imposters = new List<Imposter>();
         }
-
+        
         /// <summary>
-        /// Submits all pending imposters to be created in mountebank. Will throw a MountebankException
-        /// if unable to create the imposter for any reason.
+        /// Submits all pending imposters from the supplied collection to be created in mountebank. 
+        /// Will throw a MountebankException if unable to create the imposter for any reason.
         /// </summary>
-        public void Submit()
+        public void Submit(ICollection<Imposter> imposters)
         {
-            foreach (var imposter in Imposters.Where(imp => imp.PendingSubmission))
+            foreach (var imposter in imposters.Where(imp => imp.PendingSubmission))

```

Comment:
I'm not sure the PendingSubmission property really has any value anymore. Since the only way imposters will get into the collection maintained by the client is if they have been submitted, that property no longer tells you anything interesting. You could probably just remove it from the Imposter.